quinn-udp/src/windows.rs

```diff
@@ -107,23 +107,34 @@ impl UdpSocketState {
             )?;
         }
 
-        // Opportunistically try to enable GRO
-        _ = set_socket_option(
-            &*socket.0,
-            WinSock::IPPROTO_UDP,
-            WinSock::UDP_RECV_MAX_COALESCED_SIZE,
-            // u32 per
-            // https://learn.microsoft.com/en-us/windows/win32/winsock/ipproto-udp-socket-options.
-            // Choice of 2^16 - 1 inspired by msquic.
-            u16::MAX as u32,
-        );
-
         let now = Instant::now();
         Ok(Self {
             last_send_error: Mutex::new(now.checked_sub(2 * IO_ERROR_LOG_INTERVAL).unwrap_or(now)),
         })
     }
 
+    /// Enable or disable receive offloading.
+    ///
+    /// Also referred to as UDP Receive Segment Coalescing Offload (URO) on Windows.
+    ///
+    /// <https://learn.microsoft.com/en-us/windows-hardware/drivers/network/udp-rsc-offload>
+    ///
+    /// Disabled by default due to <https://github.com/quinn-rs/quinn/issues/2041>.
+    pub fn set_gro(&self, socket: UdpSockRef<'_>, enable: bool) -> io::Result<()> {
```

Comment:
Should we also expose this method on other platforms, just with a different default? Platform-specific APIs are a bit of a footgun.

Reply:
> Platform-specific APIs are a bit of a footgun.

I see your point. That said, the following arguments made me choose a platform specific API:

- What would `set_gro` do on platforms that don't do receive offloading, e.g. MacOS, BSDs, ...? In my eyes e.g. adding a no-op `set_gro` is a larger inconsistency (or surprise) than a platform specific API.

- I am not aware of any issues related to GRO on Linux and Android. Why offer and more importantly maintain a `fn` without a use-case?

Given the subtle issues with https://github.com/quinn-rs/quinn/issues/2041 across a wide variety of machines, I don't expect `set_gro` to be used by anyone but a tiny (brave) minority of folks that either control their hardware all the way, or implement a sophisticated heuristic to disable it programmatically.

@Ralith with the above in mind, do you still want me to add `set_gro` to the other platforms?

Reply:
> In my eyes e.g. adding a no-op set_gro is a larger inconsistency (or surprise) than a platform specific API. [...] Why offer and more importantly maintain a fn without a use-case?

I don't think this is much different from e.g. the `ecn` and `src_ip` fields of `Transmit`, which are silent no-ops on some platforms. The intent is that you can write platform-agnostic code against a fixed API and get the best possible behavior for each platform. Calling `set_gro(true)` and not getting GRO isn't harmful, or even reliably detectable.

For completeness, note one use-case for disabling GRO exists: Wireshark [still cannot handle](https://gitlab.com/wireshark/wireshark/-/issues/19109) GRO reliably, seemingly due to limitations in the kernel packet capture interface. Users trying to gather useful packet captures may wish to disable GSO/GRO to mitigate this. This comes up very rarely in practice, though. 

> I don't expect set_gro to be used by anyone but a tiny (brave) minority

That's a fair point. Idiomatic use as outlined above probably shouldn't ever call this API in the first place, since it's generally quinn-udp's responsibility to do the right thing by default. On that basis I'm happy to defer to your judgement.

Reply:
> > In my eyes e.g. adding a no-op set_gro is a larger inconsistency (or surprise) than a platform specific API. [...] Why offer and more importantly maintain a fn without a use-case?
> 
> I don't think this is much different from e.g. the `ecn` and `src_ip` fields of `Transmit`, which are silent no-ops on some platforms.

That is a good point!